### PR TITLE
JS Capstone - Deployment

### PR DESCRIPTION
**Pull request by nelsinofrancisco**
# JS Capstone

In this project, I'm supposed to choose an external API and work with it, in my case I choose tvmaze.com/api. 
I'm allowed by the Student Success team to be working alone and only doing the kanban board tasks of Student B. 

## Requirements implemented:

- Use git-flow¹
- Use Webpack and Jest libraries
- Build Interface for the basic shared UI between students
- Build Interface for the modal popup
- Test the function that counts the comments on the page
- Add a descriptive README.md file and record a video documenting the project (It's in the README.md File)

## Learning Objectives

- Use JavaScript to make websites dynamic and build basic single-page apps.
- **Use ES6 syntax.** ²
- Use ES6 modules.
- Use callbacks and promises.
- Use webpack.
- Apply JavaScript best practices and language style guides in code.
- Use AAA pattern for unit tests.
- Write units tests for a JavaScript app.
- Follow Gitflow.
- Solve simple git conflicts.
- Send and receive data from an API.
- Use API documentation.
- Understand and use JSON.
- Make JavaScript code synchronous.
- Perform a code review for a team member.

## Observations

¹ I did use the correct gitflow principle to do a PR for every task that I had to build, but since I was allowed to work solo because of some problems that my partner had In this Repo I don't have all the branch logs.  
² The First Reviewer asks me to use ES6 arrow functions or what would be lambda expressions in Java, but in the project requirements there's no mention of that, besides the learning objectives require me to use ES6 syntax, however, the class keyword is an ES6 feature...

### CODE REVIEW

File: src/modules/ModalInterface.js [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+        const result = [{
+          username: '',
+          creation_date: '',
+          comment: 'No comments Yet',
+        }];
+        return result;
+      }
+      return response.json();
+    });
+
```
Comment (oxenprogrammer) [inline src/modules/ModalInterface.js]: **Changes Required**
- [x] Fair effort on using the `async await`  promise syntactic sugar. However, you're mixing two syntax here, and the `async` keyword in this context is redundant. May we please remove the `.then` and stick to the `async await` format.

File: src/modules/ModalInterface.js [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+        const commentTextArea = addNewCommentForm.children[1].value;
+        Modal.postComment(showIndex,
+          nameInput, commentTextArea)
+          .then(() => {
+            Modal.displayComments(null, showIndex);
+          });
+        e.preventDefault();
+      });
+    }
+  }
```
Comment (oxenprogrammer) [inline src/modules/ModalInterface.js]: **Changes Required**
- [x] Fair effort on using the `async await`  promise syntactic sugar. However, you're mixing two syntax here, and the `async` keyword in this context is redundant. May we please remove the `.then` and stick to the `async await` format.

File: src/modules/UI.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        image.addEventListener('click', () => {
+          Modal.displayModal(image.dataset.showId);
+        });
+      });
+    }
+  }
+
+  static commentsListener() {
+    const comments = document.querySelectorAll('.comments-btn');
+    if (comments) {
```
Comment (oxenprogrammer) [inline src/modules/UI.js:29]: **Changes Required**
- [ ] Thanks for implementing these functionality. However, I suggest we should prefer the use of  `ES6 Arrow` functions. Let us try and refactor these functions using the `ES6 Arrow Functions.

File: README.md [hunk 4]
```diff
@@ -1 +1,92 @@
...
+- Html
+- Css
+
+## Live Demo
+
+[Live Demo Link](https://nelsinofrancisco.github.io/JSCapstone/)
+
+## Project Description (Loom Video)
+
+Check out the [video](https://www.loom.com/share/cfc64e65a3af494998491ee142811a67)
```
Comment (oxenprogrammer) [inline README.md]: **Changes Required**
- [x] Good presentation, I love your composure. However, you just missed one thing in the presentation.
**The video contains a highlight about a piece of the code the student finds interesting or is proud of making.**  _I am afraid you will have to record another video._

**Review by oxenprogrammer — CHANGES_REQUESTED** [review]
Hi @nelsinofrancisco , 

Good job so far!
**To Highlight**
- [x] Gitflow correctly followed.
- [x] Good looking UI. Good color choice.
- [x] Professional Repo.
- [x] Passing Unit Tests.
- [x] Popup works nicely.
- [x] Good communication.
    
I can see you're really putting effort to become a world-class software developer :+1:
There are some issues that you still need to address in order to move forward to the next project.
Trust me you got this! :muscle:


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Consider the optional changes nice to have features and only work on them when you have satisfied the required changes_

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: src/modules/UI.js (hunk 3 above)
Comment (nelsinofrancisco) [reply src/modules/UI.js:29]: This is a request about your personal preference? I'm using ES6 class syntax for about 4 weeks and you are the first TSE that asks me to change this, could you explain me better why arrox functions are required here?

Comment (nelsinofrancisco) [reply src/modules/UI.js:29]: For the future reviewer, if my previous question is not answered I'll not do this suggestion. 

- [ ] Check out my PR description there's no mention of using only ES6 arrow functions in the requirements, and if using the ES6 class syntax that's no way or point doing const var = () => {} ;

File: README.md (hunk 4 above)
Comment (nelsinofrancisco) [reply README.md]: Thank you for pointing it out, I did the change.

File: src/modules/ModalInterface.js (hunk 2 above)
Comment (nelsinofrancisco) [reply src/modules/ModalInterface.js]: In this one, I'm not really sure why I did this, but I think think that the await keyword doesn't work inside an addEventListener function, however I created a function that will handle that properly outside of this eventListener.

File: src/modules/ModalInterface.js (hunk 1 above)
Comment (nelsinofrancisco) [reply src/modules/ModalInterface.js]: Changes made.

**Review by mikenath223 — CHANGES_REQUESTED** [review]
Hi there, 

Awesome job on making the changes requested by the previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet. Let's work on those.


 ## Required Changes ♻️
_Kindly check the comments under the review._


Happy coding, Cheers!🚀🚀

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, whether it's valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mikenath223 — CHANGES_REQUESTED** [review]
.

File: README.md [hunk 5]
```diff
@@ -1 +1,92 @@
...
+- Css
+
+## Live Demo
+
+[Live Demo Link](https://nelsinofrancisco.github.io/JSCapstone/)
+
+## Project Description (Loom Video)
+
+Check out the [video](https://www.loom.com/share/9c979ec4ed3a4784b4cafb54013640b7)
+
```
Comment (mikenath223) [inline README.md:27]: - [x] As requested in the previous review, in the video you should show and speak about a piece of code that you find interesting. Kindly record a new video doing that.

Comment (nelsinofrancisco) [reply README.md:27]: I've done that, multiples times actually, but check out in my video the minute 3:16. I know that it's a long video, but I was talking about a part of my code function() that I enjoy doing and what the function is doing too.

Comment (henatan99) [reply README.md:27]: This issue is well addressed. I also commend you for effective communication by mentioning the time in the video I should look at. :+1:

**Review by henatan99 — APPROVED** [review]
Hi @nelsinofrancisco , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by nelsinofrancisco** [comment]
Hello @henatan99 @oxenprogrammer @mikenath223,

Thank you for your time and effort, I'll merge my project now!

**Pull request merged**